aws/delete_s3_velero_bucket.py

```diff
@@ -0,0 +1,233 @@
+"""
+The main purpose of this script is to delete a bucket associated with a cluster
+after the cluster is uninstalled:
+-   The script will delete the velero bucket that is mapped to a cluster.
+
+Required (environment Variable):
+    AWS_ACCESS_KEY_ID
+    AWS_SECRET_ACCESS_KEY
+
+Usage:
+1. The module can be imported and used like below :
+
+    import delete_s3_velero_bucket
+
+    delete_s3_bucket.delete_velero_cluster_buckets(cluster)
+    -   cluster: the name of the cluster
+
+2.  python delete_s3_velero_bucket.py -c|--cluster <cluster_name>
+
+    example:
+    1. python delete_s3_velero_bucket.py -c mts-rhods-c1
+    2. python delete_s3_velero_bucket.py --cluster mts-rhoam-c2
+
+"""
+import argparse
+import json
+import logging
+import os
+import re
+import sys
+from http import HTTPStatus
+from typing import Union
+
+import boto3
+
+LOGGER = logging.getLogger(__name__)
+S3_CLIENT = boto3.client("s3")
+REQUIRE_ENV = ["AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY"]
+
+
+def delete_velero_cluster_buckets(cluster) -> None:
+    """
+    Delete the velero bucket associated with a cluster
+
+    Args:
+        cluster: The cluster name
+
+    """
+
+    LOGGER.info(f"Delete velero buckets for '{cluster}' cluster")
+
+    velero_bucket_list = get_velero_buckets()
+    if not velero_bucket_list:
+        LOGGER.info("No velero buckets found")
+        return
+
+    for bucket in velero_bucket_list:
+        if verify_cluster_matches_velero_infrastructure_name(
+            cluster_name=cluster,
+            bucket_name=bucket["Name"],
+        ):
+            delete_all_objects_from_s3_folder(
+                bucket_name=bucket["Name"],
+            )
+
+            delete_bucket(bucket_name=bucket["Name"])
+            # A cluster is mapped to only one bucket, so don't check any remaining buckets
```

Comment:
in this case don't you want to check that `len(velero_bucket_list) == 1` ?

Reply:
it done in: line 52
velero_bucket_list = get_velero_buckets()
    if not velero_bucket_list:
        LOGGER.info("No velero buckets found")
        return

Reply:
I mean, you exit here inside the loop after the first match, but what if you have more than one bucket in the list?

Reply:
Velero bucket are created as backup for the cluster meta date, we will have one bucket for cluster. We loop to file the bucket by cluster name, clean the objects and delete it.